conformance/lang/expressions/logical-expr/logical_or_expr.balt

```diff
@@ -0,0 +1,853 @@
+    boolean _ = 23.toBalString() || 23.toBalString(); // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using error constructor expression.
+Labels: logical-expr, error-constructor-expr
+
+function errorFunction() {
+    boolean _ = true || error("Message"); // @error static type of each expression must be a subtype of boolean
+    boolean _ = error("Message") || true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = error("Message") || error("Message"); // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using anonymous function expression.
+Labels: logical-expr, explicit-anonymous-function-expr, string, additive-expr
+
+function errorFunction() {
+    boolean _ = true || function (string x, string y) returns string {return x + y;}; // @error static type of each expression must be a subtype of boolean
+    boolean _ = function (string x, string y) returns string {return x + y;} || true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = function (string x, string y) returns string {return x + y;} || function (string x, string y) returns string {return x + y;}; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using type cast expression.
+Labels: logical-expr, type-cast-expr, float
+
+function errorFunction() {
+    boolean _ = true || <float>12; // @error static type of each expression must be a subtype of boolean
+    boolean _ = <float>12 || true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = <float>12 || <float>12; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using typeof expression.
+Labels: logical-expr, typeof-expr
+
+function errorFunction() {
+    boolean _ = true || typeof 12; // @error static type of each expression must be a subtype of boolean
+    boolean _ = typeof 12 || true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = typeof 12 || typeof 12; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using typeof expression.
+Labels: logical-expr, typeof-expr
+
+function errorFunction() {
+    boolean _ = true || typeof true; // @error static type of each expression must be a subtype of boolean
+    boolean _ = typeof true || true; // @error static type of each expression must be a subtype of boolean
```

Comment:
Duplicate of previous test?

Reply:
[94d5da7](https://github.com/ballerina-platform/ballerina-spec/pull/1113/commits/94d5da7d6a3c741d8d2a8cb2d22ac6910b20f776)